globus_automate_client/flows_client.py

```diff
@@ -311,28 +320,55 @@ def list_flows(
             where the retrieving identity has at least one of the listed roles on
             each Flow
         """
-        self.authorizer = get_authorizer_for_scope(MANAGE_FLOWS_SCOPE, self.client_id)
+        self.authorizer = self.flow_management_authorizer
         params = {}
         if roles is not None and len(roles) > 0:
             params.update(dict(roles=",".join(roles)))
         return self.get("/flows", params=params, **kwargs)
 
-    def _scope_for_flow(self, flow_id: str) -> str:
-        flow_defn = self.get_flow(flow_id)
-        flow_scope = flow_defn.get("globus_auth_scope", flow_defn.get("scope_string"))
-        return flow_scope
+    def delete_flow(self, flow_id: str, **kwargs):
+        """
+        Remove a Flow definition and its metadata from the Flows service
 
-    def _action_client_for_flow(
-        self, flow_id: str, flow_scope: Optional[str] = None
-    ) -> ActionClient:
+        :param flow_id: The UUID identifying the Flow to delete
+        """
+        self.authorizer = self.flow_management_authorizer
+        return self.delete(f"/flows/{flow_id}", **kwargs)
+
+    def scope_for_flow(self, flow_id: str) -> str:
+        """
+        Returns the scope associated with a particular Flow
+
+        :param flow_id: The UUID identifying the Flow's scope to lookup
+        """
+        flow_url = f"{self.base_url}/flows/{flow_id}"
```

Comment:
I wonder if we can try a simple unauthenticated GET here first so that in the case (not sure how common) where the lookup can be done without authentication, we avoid having to make the user go through the consent flow or more specifically here give us back an authenticator.

Reply:
From my testing with this, it didn't look like we could introspect a Flow to pull it's scope without authentication: 
`
code | "AUTHENTICATION_ERROR"
detail | "Authorization header missing"
`
Is this supposed to be/ok to be public info? I was using the authorizer created with the MANAGE_SCOPES grant to do this and that seemed to be working fine (but adds some juggling of authorizers).

Reply:
I think you're probably right.

We should probably in the Flows service be looking at the value the user provided on `visible_to` and allow the unauthenticated case. But, that's a server-side problem first, then we can address here. So, probably leave as is for now.